pkgs/applications/virtualization/sail-riscv/default.nix

```
@@ -0,0 +1,48 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+
+, arch
+
+, ocamlPackages
+, ocaml
+, zlib
+, z3
+}:
+
+
+stdenv.mkDerivation rec {
+  pname = "sail-riscv";
+  version = "0.5";
+
+  nativeBuildInputs = with ocamlPackages; [ ocamlbuild findlib ocaml z3 ];
+  buildInputs = with ocamlPackages; [ sail zlib ];
+
+  prePatch = "rm -r prover_snapshots";
+
+  src = fetchFromGitHub {
+    owner = "riscv";
+    repo = pname;
+    rev = version;
+    hash = "sha256-7PZNNUMaCZEBf0lOCqkquewRgZPooBOjIbGF7JlLnEo=";
+  };
+
+  makeFlags = [ "SAIL=sail" "ARCH=${arch}" "SAIL_DIR=${ocamlPackages.sail}/share/sail" ];
```

Keep them one per line. This would make future modifications easier and cleaner.
And is sail used as a tool during build? That way it should be in nativeBuildInputs.
```suggestion
  makeFlags = [
    "SAIL=sail"
    "ARCH=${arch}"
    "SAIL_DIR=${ocamlPackages.sail}/share/sail"
  ];
```